### ISSUE TITLE
Showing only depatures 5 hours ahead

### ISSUE THREAD

**Issue by Valappi** [vingerha/gtfs2#100]
I did a install a the GTFS2 integration and got it working last night. 
It seemed to work as expected with real-time data updates and everything and showing upcoming departures within the hour.

Today when I revisit the install, it only displays departures 5-6h in the future and have been doing so constantly for 7h+.
So when the time was 10:00, it displayed departures around 15:00 and the sensor saying the same thing ("In 5 hours").
And now at around 16:00, it displays departures around 21:00 and sensor saying "In 5 hours".

Have rebooted the whole thing, set up new tram routes with the same issue still occurring. 
Not occurring on bus routes though!

Any idea in which end to start?

- important: url to the zip file of the **static datasource**
https://www.trafiklab.se/sv/api/trafiklab-apis/gtfs-regional/ , the datasource in question is "behind" api-key.
Can supply a link in private of the static datasource. 

- important if realtime issue: url to the source too
Same here as above. Today it doesn't display any Next depature real-time info (only "-") but the " Gtfs rt updated at" field keeps updating. (And also "Gtfs updated at")

- route ID
- stop ID
- outward/return
Can supply this info in private.

**Release used**
- gtfs2 release 
0.4

- HA type (HAOS/Container)
HAOS, Yellow

**Additional**
Please add logs if helpfull, switch to 'debug' logging to get more details

**Comment by vingerha** [comment]
I read that when you installed it, it showed trips/departures without a time-delay
There is no reason why this would change overnight 

Options for a delayed display 
- you have configured it with an offset (in : options), I guess you did not do this (almost no one does)
- the static datasource is providing these differences, e.g. the time might not comply with your timezone
- you have HA configured for a TZ different then the one of the datasource, e.g. you are in USA but using data from Sweden
I cannot see any of those in your issue above but I also donot have the required details to set this up so can not help further, in the past I did test 2/3 SE based trafiklab routes and they were fine

Wrt to realtime... this would be expected as RT is usually not provided for trips being far in the future, i.e. the bus/tram has not even started to move yet.

**Comment by Valappi** [comment]
>There is no reason why this would change overnight

It baffles me as well.

- you have configured it with an offset (in : options), I guess you did not do this (almost no one does)
No offset was set initially. 
I set 1 minute today in an attempt to see if anything happened and later on returned it to 0, nothing changed.

- the static datasource is providing these differences, e.g. the time might not comply with your timezone
To be honest, I don't really trust that the transit company that provides the datasource hasn't done anything stupid with it but I found. what I assume is data from the same source, on moovitapp.com yesterday that included all of the scheduled trams during the day. And it feels weird that the other means of transportations have an correct, working timetable in the dataset.

- you have HA configured for a TZ different then the one of the datasource, e.g. you are in USA but using data from Sweden
Not applicable sadly. 

I use the same data source for bus as tram and the bus sensor has been working the whole day today without a issue.
What I did notice yesterday evening though was that the closer I came to the original time I set the sensor up the previous day, it begun working. 

Times are an estimate now but say I configured it 22:00 and the sensor worked and showed me the time until the next departure as expected. 
The next day, the sensor showed "In 6 hours" prior 16:00. 
The closer the time gets to when I originally configured it (22:00) it starts counting down.
So at 18:00 the sensor shows "in 4 hours"
At 20:00 the sensor shows "in 2 hours" 
After 22:00 the sensor shows the correct time to the next departure and does so up until (I assume), 00:00

So for some reason is seems that the sensor is stuck on the time I set it up! 

I tried to set up a new sensor for the same route now but it only states "This entity is not available".
I have 5 disabled sensors (the ones set up originally) and same thing.
Deleted 3 of them and tried to set up a new one and still get "This entity is not available". 
So now I can't set up new sensors at all. Is there a limit to number of sensors?
